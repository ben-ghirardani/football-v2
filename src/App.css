/* page wasn't taking up full screen so added height 100% to body and html */
body, html {
  height: 100%;
  margin: 0px;
  padding: 0px;
}

.app {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  background-image: url("./images/GrassBackground.png");
  background-repeat: no-repeat;
  background-size: cover;
}