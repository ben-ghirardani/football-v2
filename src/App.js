import React, { Component } from 'react';
import authToken from './AuthToken';
import Loading from './components/loading/Loading';
import { BrowserRouter as Router, Route} from 'react-router-dom';
import MainPage from './components/main_page/MainPage.js'
import Matches from './components/matches/Matches.js'
import './App.css'

export default class App extends Component {

  constructor(props) {
    super(props);
      this.state = {
        currentTeam: null,
        error: false,
        loading: true,
        matches: null,
        standings: null,
        teams: null,
      }
      this.fetchStandings = this.fetchStandings.bind(this);
      this.fetchMatches = this.fetchMatches.bind(this);
      this.fetchTeams = this.fetchTeams.bind(this);
      this.updateCurrentTeam = this.updateCurrentTeam.bind(this);
  }

  componentDidMount() {
    this.fetchStandings();
    this.fetchMatches();
    this.fetchTeams();
  }

    // is this complete? error handling?
  async fetchStandings() {
    if (this.state.standings === null) {
      const response = await fetch(`http://api.football-data.org/v2/competitions/2021/standings`, {headers : {'X-Auth-Token': authToken}});
      const data = await response.json();
      this.setState({standings: data, loading: false});
    } else {
      return
    }
  }

  async fetchMatches() {
    if (this.state.matches === null) {
      const response = await fetch(`http://api.football-data.org/v2/competitions/2021/matches`, {headers : {'X-Auth-Token': authToken}});
      const data = await response.json();
      this.setState({matches: data});
    } else {
      return
    }
  }

  async fetchTeams() {
    if (this.state.teams === null) {
      const response = await fetch(`http://api.football-data.org/v2/competitions/2021/teams`, {headers : {'X-Auth-Token': authToken}});
      const data = await response.json();
      this.setState({teams: data});
    } else {
      return
    }
  }

  // fetchStandings() {
  //   fetch(`http://api.football-data.org/v2/competitions/2021/standings`, {headers : {'X-Auth-Token': authToken}} )
  //     .then(response => response.json())
  //     .then(data => this.setState({standings: data, loading: false}))
  //     .catch(error => this.setState({ error: false }));
  // }	

  updateCurrentTeam(team) {
    this.setState({currentTeam: team});
  }

  render() {
    return (
      <div className="app">
        {
          this.state.loading ? <Loading/> : 
            <>
          <Router>
            <>
              <Route path="/" exact render=
                {() => 
                  <MainPage
                    standings={this.state.standings}
                    updateCurrentTeam={this.updateCurrentTeam} 
                  />
                }
              />
              <Route path="/fixtures/" render=
                {() => 
                  <Matches
                    currentTeam={this.state.currentTeam} 
                    matches={this.state.matches}
                  />
                }
              />
            </>
          </Router>
          
      </> 
        }
      </div>
    )
  }

}