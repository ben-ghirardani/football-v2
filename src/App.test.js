import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
// import "isomorphic-fetch";
import ReactDOM from 'react-dom';
import App from './App';
import { jsxEmptyExpression } from '@babel/types';

it('matches snapshot', () => {
  const wrapper = shallow(<App/>);
  expect(wrapper).toMatchSnapshot();
});

