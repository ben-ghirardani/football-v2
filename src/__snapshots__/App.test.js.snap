// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `ShallowWrapper {}`;

exports[`renders without crashing 1`] = `ShallowWrapper {}`;
